scripts/refcodes_with_properties/refcodes_with_properties.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+#
+# This script can be used for any purpose without limitation subject to the
+# conditions at http://www.ccdc.cam.ac.uk/Community/Pages/Licences/v2.aspx
+#
+# This permission notice and the following statement of attribution must be
+# included in all copies or substantial portions of this script.
+#
+# 2025-03-14: created by Jason C. Cole, The Cambridge Crystallographic Data Centre
+
+'''
+Filter a refcode list to the subset that have the desired properties
+'''
+
+#########################################################################
+
+import sys
+import os
+import argparse
+import EntryPropertyCalculator
+from ccdc import io
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser( description=__doc__,
+                                      formatter_class=argparse.RawDescriptionHelpFormatter )
+
+
+    parser.add_argument( '-r','--refcode_file', help='input file containing the list of refcodes', default = None )
+    parser.add_argument( '-d','--database_file', help='input file containing the list of refcodes', default = None )
+    parser.add_argument( '-c','--control_file', help='configuration file containing the desired properties\n\n %s' % (EntryPropertyCalculator.helptext()) )
+    parser.add_argument( '-v','--get_values', action="store_true", help='calculate and print descriptor values where possible rather than filter\n\n %s' % (EntryPropertyCalculator.helptext()) )
+    parser.add_argument( '-o','--output_file', default = None, help='output CSV file for results\n\n %s' % (EntryPropertyCalculator.helptext()) )
+
+    args = parser.parse_args()
+
+    refcode_file = args.refcode_file
+    database_file = args.database_file
+    control_file = args.control_file
+    print_values = args.get_values
+
+    outfile = sys.stdout
+    if args.output_file != None:
+        outfile = open(args.output_file, 'wb')
+
+    filterer = EntryPropertyCalculator.parse_control_file(open(control_file,"r").readlines())
+
+    reader = None
+    if refcode_file == None:
+        if database_file == None:
+            reader = io.EntryReader('CSD')
+        else:
+            reader = io.EntryReader(database_file)
+    else:
+        reader = io.EntryReader(refcode_file, format='identifiers')
```

Comment:
Stylistic suggestion to remove nested if statements. 

```suggestion
    reader = None
    if refcode_file:
        reader = io.EntryReader(refcode_file, format='identifiers')
    elif database_file:
        reader = io.EntryReader(database_file)
    else:
        reader = io.EntryReader('CSD')
```